Cargo.lock

```diff
@@ -13,9 +13,9 @@ name = "abstract-domain-derive"
 version = "0.1.0"
 dependencies = [
  "move-stackless-bytecode",
- "proc-macro2 1.0.76",
- "quote 1.0.35",
- "syn 2.0.48",
+ "proc-macro2",
+ "quote",
+ "syn 1.0.109",
```

Comment:
Why are we rolling back to such an old version here?

It seems to be required by a diem example.
```
third_party/move/move-examples/diem-framework/crates/crypto-derive/Cargo.toml:syn = { version = "1.0.64", features = ["derive"] }
```
Can we exclude the move-examples from this workspace requirement?

Reply:
Because that is the current version which the workspace uses. From main: https://github.com/aptos-labs/aptos-core/blob/77e1d222ebc5e7294e115e0d090c001da1d0e072/Cargo.toml#L695

I think its too risky to upgrade crates as part of this PR, and in this case, the API is also different, that means changes somewhere in aptos-core would be needed.